### PR TITLE
Adds Identity initializer

### PR DESCRIPTION
**Pull request by joaogui1**
This PR adds the initializers that sonnets has but are missing in Haiku.
I haven't written tests for them yet since I don't know if there is interest in this PR, as soon as the Haiku team gives me the green light I will add the tests

### CODE REVIEW

File: haiku/_src/initializers.py [hunk 1]
```diff
@@ -254,3 +254,42 @@ def __call__(self, shape: Shape, dtype: DType) -> jnp.ndarray:
...
+    """Initializer that generates the identity matrix.
+    Constructs a 2D identity matrix or batches of these.
+    """
+
+    def __init__(self, gain: float = 1.0):
+      """Constructs an identity initializer.
+      Args:
+         gain: Multiplicative factor to apply to the identity matrix.
+      """
+    self.gain = gain
```
Comment (tomhennigan) [inline haiku/_src/initializers.py:282]: Indent plz 😄

File: haiku/_src/initializers.py [hunk 2]
```diff
@@ -254,3 +254,42 @@ def __call__(self, shape: Shape, dtype: DType) -> jnp.ndarray:
...
+    """Initializer that generates tensors initialized to 0."""
+
+    def __call__(self, shape: Shape, dtype: DType) -> jnp.ndarray:
+      return jnp.zeros(shape, dtype)
+
+  class Ones(Initializer):
+    """Initializer that generates tensors initialized to 1."""
+
+    def __call__(self, shape: Shape, dtype: DType) -> jnp.ndarray:
+      return jnp.ones(shape, dtype)
```
Comment (tomhennigan) [inline haiku/_src/initializers.py:269]: I think for Zeros and Ones it is probably more readable to just use `jnp.zeros` and `jnp.ones`. These actually match the `Initializer` signature so type checking works too 😄

File: haiku/_src/initializers.py [hunk 3]
```diff
@@ -254,3 +254,42 @@ def __call__(self, shape: Shape, dtype: DType) -> jnp.ndarray:
...
+
+    def __init__(self, gain: float = 1.0):
+      """Constructs an identity initializer.
+      Args:
+         gain: Multiplicative factor to apply to the identity matrix.
+      """
+    self.gain = gain
+
+    def __call__(self, shape: Shape, dtype: DType) -> jnp.ndarray:
+      if len(shpe) < 2:
```
Comment (tomhennigan) [inline haiku/_src/initializers.py:285]: Typo here.

File: haiku/_src/initializers.py (hunk 2 above)
Comment (tomhennigan) [reply haiku/_src/initializers.py:269]: (In Sonnet we used a type for `Initializer` while in Haiku we just used `Callable[.., ..]` so any callable with the right signature is a valid init function.)

File: haiku/_src/initializers.py (hunk 1 above)
Comment (joaogui1) [reply haiku/_src/initializers.py:282]: Still getting the hang of emacs haha

File: haiku/_src/initializers.py (hunk 2 above)
Comment (joaogui1) [reply haiku/_src/initializers.py:269]: So I remove them?

Comment (tomhennigan) [reply haiku/_src/initializers.py]: Thanks, looks good now.

**Comment by tomhennigan** [comment]
Thanks, I think including Identity is a useful addition, please add a test (you should be able to port some over from Sonnet 2).

**Comment by joaogui1** [comment]
I was starting to use emacs and it's "eletric ident" seems to have messed stuff up, sorry about that

**Comment by joaogui1** [comment]
It finally works @tomhennigan

**Comment by tomhennigan** [comment]
Sorry for the delay @joaogui1, I've been less productive than usual the last few weeks.. Looks good!

**Comment by joaogui1** [comment]
No problem, world is pretty messed up right now :(
Hey, can you tell me what internal test failed?

**Comment by tomhennigan** [comment]
It was `pytype` (gain was marked as `float` but we pass a `jnp.ndarray` in one of the tests. I've fixed this as part of the process of importing the change back into the DeepMind codebase. All the tests are passing now and I fixed another minor linter issue (adding a new line in the docstrings).

I'll see if we can add this as a github action too, that and pylint are two tools where internally we treat warnings as errors to try and have high code quality.

FYI we have a slightly janky (same as Google) setup where we mirror our internal version of Haiku to GitHub and this means to merge PRs we pull the same change into our internal repo, merge it there and then a copybara robot marks the PR as merged). That's why it takes a bit of time between approving the PR and it being merged. This should be merged in a few hours once someone internally LGTMs my proposal to import this 😄

**Comment by joaogui1** [comment]
So I guess you guys merged it, closing this PR :D

**Pull request closed without merging**